tdp/core/dag.py

```diff
@@ -45,57 +45,86 @@
 }
 DEFAULT_SERVICE_PRIORITY = 99
 
-DAG_FOLDER_NAME = "tdp_lib_dag"
-DAG_EXTENSION = ".yml"
-
 
 class Dag:
     """Generate DAG with components dependencies"""
 
-    def __init__(self, yaml_files, playbooks_dir=None):
+    def __init__(self, collections):
+        """
+        :param collections: ordered mapping of collections, with names used as keys
+        :type collections: OrderedDict[str, Collection]
+        """
+        self._collections = collections
         self._components = None
         self._graph = None
         self._yaml_files = None
         self._services = None
         self._services_components = None
-        self.yaml_files = yaml_files
-        self.playbooks_dir = playbooks_dir
 
     @staticmethod
-    def from_collection(collection_path):
-        return Dag(list((Path(collection_path) / DAG_FOLDER_NAME).glob("*" + DAG_EXTENSION)))
+    def from_collection(collection):
+        """Factory method to build a dag from a single collection. Leniant on input type
+
+        :param collection: one collection
+        :type collection: Union[str, Path, Collection]
+        :raises ValueError: if invalid type
+        :return: Dag built from input
+        :rtype: Dag
+        """
+        if isinstance(collection, (str, Path)):
+            return Dag.from_collections([Collection.from_path(collection)])
+        elif isinstance(collection, Collection):
+            return Dag.from_collections([collection])
+        raise ValueError("collection must be either an str, a Path or a Collection")
 
-    @property
-    def yaml_files(self):
-        if self._yaml_files is not None:
-            return self._yaml_files
-        return []
+    @staticmethod
+    def from_collections(collections):
+        """Factory method to build a dag from multiple collections
 
-    @yaml_files.setter
-    def yaml_files(self, value):
-        self._yaml_files = value
-        del self.components
+        Ordering of the sequence is what will determine the loading order of the components.
 
-    @yaml_files.deleter
-    def yaml_files(self):
-        self.yaml_files = None
+        :param collections: Ordered Sequence of Collection
+        :type collections: Sequence[Collection]
+        :return: Dag built from x collections
+        :rtype: Dag
+        """
+        collections = OrderedDict(
+            (collection.name, collection) for collection in collections
+        )
+
+        return Dag(collections)
+
+    @property
+    def collections(self):
```

Comment:
I added a setter in the last commit. While I don't really want a setter on `collections`, I can see the value it brings to modify it.

But, I think we shouldn't put a `deleter` on collections. Either you get it, or you modify it, but you cannot remove it. It would just break the Dag class.

Reply:
Yes no `deleter`